### DISCUSSION TITLE
Quality CLI option not working

### DISCUSSION THREAD

**Discussion by Petah** [ImageMagick/ImageMagick#5175] [Help] [answered]
Hi,

When I try to run `convert` with `-quality` I am getting an error:

```
% magick convert IMG_5258.jpg -quality 80 test.jpg
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/738.
convert: unable to open image '80': No such file or directory @ error/blob.c/OpenBlob/3533.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/738.
```

It works fine without the `-quality`, and other options also work fine.

Any ideas?

```
% magick -version
Version: ImageMagick 7.1.0-33 Q16-HDRI x86_64 20040 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(5.0) 
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps raw tiff webp xml zlib
Compiler: gcc (4.2)
```

![image](https://user-images.githubusercontent.com/134799/170577585-d6269559-96f6-4ce3-ac96-92de71cd612f.png)

**Comment by fmw42** [comment, marked as answer]
Post your original input JPG

Try proper IM 7 syntax without the convert:

`magick IMG_5258.jpg -quality 80 test.jpg`

**Reply by Petah** [reply]
Its this, but it happens with any image.

![5x7-test-1](https://user-images.githubusercontent.com/134799/170578919-d3ee9516-69c4-44e3-bf56-2c6a06a8b67b.png)

**Reply by fmw42** [reply]
What is your OS?

**Reply by fmw42** [reply]
magick 170578919-d3ee9516-69c4-44e3-bf56-2c6a06a8b67b.png -quality 80 x.jpg

works fine for me on IM 7.1.0.35 Q16 HDRI.  Perhaps you have a bad install or your JPG delegate is faulty. Try upgrading to the current release.

**Reply by fmw42** [reply]
Have you tried just rebooting?

**Reply by Petah** [reply]
I'm running osx 12.3.1. It was installed via homebrew, and I tried upgrading and reinstalling. Also tried a few different file type (png, jpg, webp, heic).

But yes `magick IMG_5258.jpg -quality 80 test.jpg` worked. 

No idea why it doesn't work with `convert` though.

Thanks.

**Reply by fmw42** [reply]
The new Imagemagick 7 replace convert with magick. But all the other tools are still old IM 6 tools that have to be called from IM 7 using magick identify, or magick montage. Convert was completely replace with new code so you do not want to combine them. You can run convert or magick if you want IM 6 or IM 7 respectively. See the porting guide at https://imagemagick.org/script/porting.php

**Comment by snibgo** [comment]
The problem also occurs in v7.1.0-32, pre-built binary, on Windows 8.1:
```
f:\web\im>%IMG7%magick toes.jpg -quality 80 t.jpg

f:\web\im>%IMG7%magick convert toes.jpg t.jpg

f:\web\im>%IMG7%magick convert toes.jpg -quality 80 t.jpg
convert: unable to open image '80': No such file or directory @ error/blob.c/OpenBlob/3533.
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/738.
```
This seems to be a bug.

Removing "80" results in no error, but it should raise an error:

```
f:\web\im>%IMG7%magick convert toes.jpg -quality t.jpg
```
